.github/workflows/pr.yaml

```diff
@@ -62,15 +63,23 @@ jobs:
       arch: "amd64"
       container_image: "rapidsai/ci-conda:latest"
       run_script: "ci/build_docs.sh"
-  wheel-build:
+  wheel-build-cpp:
     needs: checks
     secrets: inherit
     uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-24.06
     with:
+      matrix_filter: group_by([.ARCH, (.CUDA_VER|split(".")|map(tonumber)|.[0])]) | map(max_by(.PY_VER|split(".")|map(tonumber)))
       build_type: pull-request
-      script: ci/build_wheel.sh
+      script: ci/build_wheel_cpp.sh
```

Comment:
*question*: My understanding of this wheel is that is morally "just" building a zip file of the RMM C++ headers, and adding some metadata so that pip can do the right thing.

It is architecture-specific because we bundle static libraries for libfmt and libspdlog, however, AFAICT, it's not cuda-specific: I downloaded the cu11 and cu12 wheels and the only difference is an 8-byte difference in the hexdump of the static libraries which corresponds to the different times they were built at:
```
diff -u <(objdump -x librmm_cu11-24.6.0a38-py3-none-any/librmm/lib64/libfmt.a) <(objdump -x librmm_cu12-24.6.0a38-py3-none-any/librmm/lib64/libfmt.a)
--- /proc/self/fd/11	2024-05-03 09:36:47.185200190 +0100
+++ /proc/self/fd/18	2024-05-03 09:36:47.185200190 +0100
@@ -1,7 +1,7 @@
-In archive librmm_cu11-24.6.0a38-py3-none-any/librmm/lib64/libfmt.a:
+In archive librmm_cu12-24.6.0a38-py3-none-any/librmm/lib64/libfmt.a:
 
 format.cc.o:     file format elf64-x86-64
-rw-r----- 0/0 227728 May  3 06:38 2024 format.cc.o
+rw-r----- 0/0 227728 May  3 06:31 2024 format.cc.o
 architecture: i386:x86-64, flags 0x00000011:
 HAS_RELOC, HAS_SYMS
 start address 0x0000000000000000
@@ -3337,7 +3337,7 @@
 
 
 os.cc.o:     file format elf64-x86-64
-rw-r----- 0/0  19376 May  3 06:38 2024 os.cc.o
+rw-r----- 0/0  19376 May  3 06:31 2024 os.cc.o
 architecture: i386:x86-64, flags 0x00000011:
 HAS_RELOC, HAS_SYMS
 start address 0x0000000000000000
```


Corollary, each of these matrix jobs takes 3 minutes (which is not a great deal in the scheme of things) but 2:30 of that is just downloading a docker container that (I think) contains much more stuff than we really need: I think we just need the rapids gha-tools and a python? So, (not here, because I suspect that refactoring the shared workflows to support this would be tremendously messy), do we care about this and should we have an option to run stripped down containers?

We could probably at least ask for a smaller node than the default 16-cpu one.

Reply:
Yeah, I brought up basically this discussion [above](https://github.com/rapidsai/rmm/pull/1529#discussion_r1588634911), and I do think it's worthwhile for us to revisit this in the future. I'm not even sure that we need to bundling the static libraries of libfmt and libspdlog since everything in RAPIDS should be using those header-only anyway. I would like to revisit this in conjunction with updating the conda packages as well since we build arch-specific conda packages for rmm too.